src/flownet/ert/_create_ert_setup.py

```diff
@@ -47,14 +48,42 @@ def create_observation_file(
         Nothing
 
     """
-    num_dates = len(schedule.get_dates())
+    # pylint: disable-msg=too-many-locals
+    dt_schedule = pd.to_datetime(schedule.get_dates())
+
+    # Resampling dates based on requested frequency with
+    # no interpolation: keeps nearest existing date
+    if config.flownet.data_source.simulation.resampling:
+        dt_resampled = pd.date_range(
+            schedule.get_dates()[1],
+            schedule.get_dates()[-1],
+            freq=config.flownet.data_source.simulation.resampling,
+        )
+        df_schedule = pd.DataFrame(data=range(len(dt_schedule)), index=dt_schedule)
+        idx = np.zeros(len(dt_resampled), dtype=int)
+        for i, k in enumerate(dt_resampled):
```

Comment:
How fast is this loop?

Reply:
it is not slow as it only concerns the dates, not all the data. I tried to avoid the for-loop, but the `get_loc` method of `pandas` isn't vectorized